src/main/java/sirius/db/es/types/DenseVector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.es.types;
+
+import sirius.db.es.NearestNeighborsSearch;
+import sirius.kernel.commons.Strings;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Specifies a dense vector to be stored in {@linkplain sirius.db.es.ElasticEntity elastic entities}.
+ * <p>
+ * Such fields are essentially vectors of floats and usually filled by performing an embedding of a text field or
+ * image data. Based on the fact that "similar" values yield similar vectors, this can be used to perform nearest
+ * neighbor searches during retrieval. Using {@link sirius.db.es.ElasticQuery#knn(NearestNeighborsSearch)} an
+ * approximate nearest neighbor search can be performed which is implemented using rather efficient HNSW graphs.
+ */
+public class DenseVector {
+
+    /**
+     * Specified the similarity function to apply.
+     */
+    public enum Similarity {
+        COSINE("cosine"), EUCLIDEAN("l2_norm"), INNER_PRODUCT("dot_product");
```

Comment:
**Optional:** Maybe shortly describe the different Valses

Reply:
…and is it good to user other names in the enum than in ES lingo?

Reply:
> …and is it good to user other names in the enum than in ES lingo?

you might want to discuss that with copilot - I'm just the guy that commited.... ;-)